### PR TITLE
feat: add pruneStaleBranches config option

### LINKED ISSUE #4106: Optional stale branch pruning
**What would you like Renovate to be able to do?**

We've setup Renovate for our BitBucket Cloud repositories, but Renovate autoclosed PR's and deleted "orphan" branches that were still in progress.

I believe this behavior is due to the expectation that branches automatically get all commits from the main development branch merged into them. ❓

With our workflow, Renovate almost accidentally deleted code it shouldn't have.

**Describe the solution you'd like**
Add a config flag `enablePruning` (default `true`) to make stale branch pruning optional. If false, skip pruning but log what would have been applied if it was enabled (same as dryRun).

**Describe alternatives you've considered**
None

**Additional context**
* We're using BitBucket Cloud
* Renovate is run on a scheduled BitBucket pipeline

### PR DESCRIPTION
**Pull request by richardcornelissen**
Makes stale branch pruning configurable
Closes #4106

### CODE REVIEW

File: lib/config/definitions.js [hunk 1]
```diff
@@ -1847,6 +1847,13 @@ const options = [
     mergeable: true,
     cli: false,
   },
+  ,
```
Comment (ViceIce) [inline lib/config/definitions.js]: remove that, that why ci is failing

File: docs/local-development.md [hunk 2]
```diff
@@ -117,7 +117,7 @@ We wish to keep backwards-compatibility as often as possible, as well as make
 the code configurable, so most new functionality should be controllable via
 configuration options.
 
-If you wish to add one, add it to `lib/config/definitions.js` and then add documentation to `website/docs/_posts/2017-10-05-configuration-options.md`.
```
Comment (ViceIce) [inline docs/local-development.md]: why that?

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by ViceIce** [comment]
@richardcornelissen Please sign the cla

File: docs/local-development.md (hunk 2 above)
Comment (rarkins) [reply docs/local-development.md]: This looks like an old link, change looks good

**Comment by ViceIce** [comment]
for codecov copy
https://github.com/renovatebot/renovate/blob/642deb0b8330a9fafdc6a1f5c80ed14eb8d01178/test/workers/repository/finalise/prune.spec.js#L46-L57

rename tewst and replace `config.dryRun = true;` with `config.pruneStaleBranches = false;`

**Pull request merged**